### PR TITLE
Only shows order that actually have at least one line_item V2

### LINKED ISSUE #5236: How does OFN count the number of incomplete orders?
## Description

I have a user with 747 uncompleted orders. He is a bit worried because he never had any real orders. So he is afraid people don't manage to order...

## Steps to Reproduce

See uncompleted orders for hub 730 in French production

## Animated Gif/Screenshot

image

## Severity
bug-s3: a feature is broken but there is a workaround

## Your Environment
* Version used: v2.9.2
* Browser name and version: Firefox 76
* Operating System and version (desktop or mobile): Desktop

### PR DESCRIPTION
**Pull request by jibees**
#### What? Why?
Filter inside the API to shows only orders that have at least one line items
Closes #5236

This PR modify the API contract.
~~This PR should be compared to #8147~~

#### What should we test?

As an admin, go to `/admin/orders`, you should not be able to see empty (ie. with no line items) orders.

#### Release notes

 - Remove empty orders from `/admin/orders` interface
 - Do not sent empty orders from `/api/v0/orders.json`
Empty orders are defined as orders that don't have at lease one item.

Changelog Category: User facing changes & Technical changes

### CODE REVIEW

**Comment by Matt-Yorkley** [comment]
We could move this logic to a scope on `Spree::Order` like this:

```ruby
# app/models/spree/order.rb

scope :not_empty, -> {
  joins(:line_items).group(:id).having("count(spree_line_items.id) > 0")
}
```
and then use it like: 
```ruby
Permissions::Order.new(current_user).editable_orders.not_empty
```

**Comment by jibees** [comment]
> ```ruby
> scope :not_empty, -> {
>   joins(:line_items).group(:id).having("count(spree_line_items.id) > 0")
> }
> ```

Nice! 👍 😊 

I've force-pushed then ;)

**Review by apricot12 — APPROVED** [review]
This looks great. 🚀

File: spec/services/search_orders_spec.rb [hunk 1]
```diff
@@ -15,7 +16,7 @@
     let(:service) { SearchOrders.new(params, enterprise_user) }
 
     it 'returns orders' do
-      expect(service.orders.count).to eq 3
+      expect(service.orders.count.length).to eq 3
```
Comment (Matt-Yorkley) [inline spec/services/search_orders_spec.rb:19]: `.count.length` looks really weird here. Shouldn't count already return `3`..?

**Comment by Matt-Yorkley** [comment]
The scope wasn't quite working as expected. I pushed a couple of commits to update it and added a unit test on it :+1:

**Review by Matt-Yorkley — APPROVED** [review]
:+1:

**Comment by filipefurtad0** [comment]
Hey @jibees ,

I've looked for orders which are not complete (ticked-off the option "Only show completed orders"), and got:

- before staging - 262 orders (1)
image

- after staging - 225 orders (2)
image

Interestingly, 3 empty orders still appear, they seem to be in `address` state (3):
image

Any idea why this happens?

(Also, it remains a mystery to me, why the two first orders on pic (1) switched state from `cart` to `address`... )

Apart from the `address` state orders: this works, let's merge :+1: 

I think merging this will raise the impact (eventually the severity) these S3s - #7637 and/or #8131, as currently the workaround is to edit the order and add items to them (so they become visible to customers and don't trigger snails to customers) or manually cancel the order. A quick demo on how bug #7637 makes it disappear from the list of orders:

Peek 2021-09-08 16-49

Anyway, looks good to go. Thank you!

**Comment by jibees** [comment]
> The scope wasn't quite working as expected. I pushed a couple of commits to update it and added a unit test on it 👍

Thanks @Matt-Yorkley : It's now more clear (both unit tests, and scope). 🙏

**Comment by jibees** [comment]
@filipefurtad0 
> Interestingly, 3 empty orders still appear, they seem to be in address state (3):

Should I debug those ones? Or should we merge?

**Comment by filipefurtad0** [comment]
Thanks for double checking @jibees .

> Should I debug those ones? Or should we merge?

My concern here is that we are introducing some behavior we're not quite sure about - which could have some unintended side effects, like hiding orders which are not empty.

I don't know how and why order states affecting this, but it seems to play a role. Perhaps best to debug?

**Comment by filipefurtad0** [comment]
It seems both these ```address``` state orders have payments associated with them. This seems to be the reason they appear on the search results, despite having zero line items - they have had line items, at some point.

However, this might not be a bad idea to keep these results, if we keep in mind that the use case is to enable hub managers to notify users on their pending orders.

I felt this was worth mentioning considering the definition: "Empty orders are defined as orders that don't have at least one item." but find no reason not to merge :+1: 


Peek 2021-09-09 11-39


However I did notice that ordering by "Total" is not really working, it needs an issue:

**Comment by filipefurtad0** [comment]
>It seems both these address state orders have payments associated with them. This seems to be the reason they appear on the search results, despite having zero line items - they have had line items, at some point.

Ok, I think I've found out what's happening here:
A customer places items and the cart, but during checkout the item goes out of stock. The order is left on the `cart` state, and only changes to `address` state once the admin clicks on it:

Peek 2021-09-13 10-25

After finding this cart/address order, the admin cannot access the payments tab:

Peek 2021-09-13 10-26

Opening issues on this. In what concerns this PR it is fine I guess that these are displayed to the admin, although these are corrupt and cannot be completed.

**Pull request merged**